File: zenscript-code-model/src/main/java/raylras/zen/model/symbol/ParseTreeLocatable.java
package raylras.zen.model.symbol;

import org.antlr.v4.runtime.tree.ParseTree;
import raylras.zen.model.CompilationUnit;

import java.nio.file.Path;

public interface ParseTreeLocatable extends Locatable {

    ParseTree getCst();

    CompilationUnit getUnit();

    default Path getPath() {
        return getUnit().getPath();
    }

    default String getUri() {
        return getPath().toUri().toString();
    }

}

File: zenscript-code-model/src/main/kotlin/raylras/zen/model/symbol/ParseTreeLocatable.kt
package raylras.zen.model.symbol

import org.antlr.v4.runtime.tree.ParseTree
import raylras.zen.model.CompilationUnit
import java.nio.file.Path

interface ParseTreeLocatable : Locatable {
    val cst: ParseTree

    val unit: CompilationUnit

    override val path: Path
        get() = unit.path
}
